compiler/surface/parser.mly

```diff
@@ -618,26 +684,31 @@ law_heading:
   }
 }
 
-law_text:
-| lines = nonempty_list(LAW_TEXT) { String.trim (String.concat "" lines) }
+let law_text :=
+| lines = nonempty_list(LAW_TEXT) ; { String.trim (String.concat "" lines) }
 
-source_file_item:
-| text = law_text { LawText text }
-| BEGIN_CODE code_and_pos = code text = END_CODE  {
+let source_file_item :=
+| text = law_text ; { LawText text }
+| BEGIN_CODE ;
+  code_and_pos = code ;
+  text = END_CODE ; {
   let (code, _) = code_and_pos in
   CodeBlock (code, (text, Pos.from_lpos $sloc), false)
 }
-| heading = law_heading {
+| heading = law_heading ; {
   LawHeading (heading, [])
 }
-| code = metadata_block {
+| code = metadata_block ; {
   let (code, source_repr) = code in
   CodeBlock (code, source_repr, true)
 }
-| BEGIN_DIRECTIVE LAW_INCLUDE COLON args = nonempty_list(DIRECTIVE_ARG) page = option(AT_PAGE) END_DIRECTIVE {
+| BEGIN_DIRECTIVE ; LAW_INCLUDE ; COLON ;
+  args = nonempty_list(DIRECTIVE_ARG) ;
+  page = option(AT_PAGE) ;
+  END_DIRECTIVE ; {
   let filename = String.trim (String.concat "" args) in
   let pos = Pos.from_lpos $sloc in
-  let jorftext = Re.Pcre.regexp "JORFTEXT\\d{12}" in
+  let jorftext = Re.Pcre.regexp "JORFTEXT\\d; {12}" in
```

Comment:
Doesn't this extra `;` break the regexp?

Reply:
Ouch, well spotted, overuse of regexp-replaces :sweat_smile: